common/src/main/java/io/netty/util/internal/StringUtil.java

```diff
@@ -309,12 +310,29 @@ public static String simpleClassName(Class<?> clazz) {
             return "null_class";
         }
 
-        Package pkg = clazz.getPackage();
-        if (pkg != null) {
-            return clazz.getName().substring(pkg.getName().length() + 1);
-        } else {
-            return clazz.getName();
+        /**
+         * Modified version of commons-lang <a href="https://github.com/apache/commons-lang/
+         * blob/LANG_3_4_RC2/src/main/java/org/apache/commons/lang3/ClassUtils.java#L194">ClassUtils</a>
+         */
+        String className = clazz.getName();
+        final StringBuilder arrayPrefix = new StringBuilder();
+
+        // Handle array encoding
+        if (className.startsWith("[")) {
+            while (className.charAt(0) == '[') {
+                className = className.substring(1);
+                arrayPrefix.append("[]");
+            }
+            // Strip Object type encoding
+            if (className.charAt(0) == 'L' && className.charAt(className.length() - 1) == ';') {
+                className = className.substring(1, className.length() - 1);
```

Comment:
Is it worth it to restructure this code to avoid the object allocation overhead and copy operations of calling `substring` repeatedly?